Travis: testing astyle job failures

0xc0170 opened the pull request:
### Description

<!-- 
    Required
    Add here detailed changes summary, testing results, dependencies 
    Good example: https://os.mbed.com/docs/latest/reference/workflow.html (Pull request template)
-->

From one branch to another,  testing travis astyle job (will delete both branches once closed)

### Pull request type

<!-- 
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!-- 
    Optional
    Request additional reviewers with @username
-->

0xc0170 committed:
test

ciarmcom commented:
@0xc0170, thank you for your changes.
@ARMmbed/mbed-os-maintainers please review.

0xc0170 commented:
@cmonr This fails as expected (reproduced on another PR targeting a branch instead of master). 

We should rather use variables travis provides? https://docs.travis-ci.com/user/environment-variables/ 

`TRAVIS_PULL_REQUEST_SHA` ?

0xc0170 committed:
travis: use fetch head for comparision

Commit:
Checking to see if the git depth param affects the build

Commit:
Check out base branch before astyle comparison

...and revert depth change.

Commit:
Fetch $TRAVIS_BRANCH before comparing it

Travis CI does not pull the base branch, only the last 50 commits.
This is fine for PRs targeting master, but fails when PRs for feature branches come in.

Commit:
Attempting a slightly different diff syntax

Doing the fetch was overriding the original `git fetch`, and would prefer not to have to run that command twice.

Commit:
Attempting to only pull the branch reference

Commit:
Added flags to `git fetch` to unshallow clone

Commit:
Checking if fetching the base branch changes anything

Commit:
Adding status command for further debugging...

Commit:
Attempting fetch vs head comparison again

cmonr commented:
Continued (and hopefully resolved?) in https://github.com/ARMmbed/mbed-os/pull/9346
